exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java

```diff
@@ -26,27 +26,37 @@
   private MockCloudMetricClient mockClient;
 
   @Before
-  public void setup() throws IOException {
-    // Find a free port to spin up our server at.
-    ServerSocket socket = new ServerSocket(0);
-    int port = socket.getLocalPort();
-    String address = String.format("%s:%d", LOCALHOST, port);
-    socket.close();
+  public void setup() throws MockServerStartupException {
+    try {
+      // Find a free port to spin up our server at.
+      ServerSocket socket = new ServerSocket(0);
+      int port = socket.getLocalPort();
+      String address = String.format("%s:%d", LOCALHOST, port);
+      socket.close();
 
-    // Start the mock server. This assumes the binary is present and in $PATH.
-    // Typically, the CI will be the one that curls the binary and adds it to $PATH.
-    String[] cmdArray = new String[] {System.getProperty("mock.server.path"), "-address", address};
-    ProcessBuilder pb = new ProcessBuilder(cmdArray);
-    pb.redirectErrorStream(true);
-    mockServerProcess = pb.start();
+      // Start the mock server. This assumes the binary is present and in $PATH.
+      // Typically, the CI will be the one that curls the binary and adds it to $PATH.
```

Comment:
This doesn't seem to assume it's in `$PATH`? Is the comment outdated?

Reply:
Yeah, I'm going to fix this up in a follow-on.  I want to use test-containers here, maybe we discuss that more offline.